packages/sit-onyx/src/components/OnyxToastProgressBar/OnyxToastProgressBar.vue

```diff
@@ -0,0 +1,53 @@
+<script lang="ts" setup>
+import { computed, watchEffect } from "vue";
+import { useTimer } from "../../composables/useTimer";
+import { injectI18n } from "../../i18n";
+import type { OnyxToastProps } from "../OnyxToast/types";
+
+const props = defineProps<Required<Pick<OnyxToastProps, "duration">>>();
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the timer has ended.
+   */
+  timerEnded: [];
+}>();
+
+const { t } = injectI18n();
+
+const { timeLeft, isEnded } = useTimer({
+  endTime: computed(() => Date.now() + props.duration),
+  useAnimationFrame: true,
+});
+
+const percentage = computed(() => {
+  if (props.duration <= 0) return 0;
+  return Math.min(100, (timeLeft.value / props.duration) * 100);
+});
+
+watchEffect(() => isEnded.value && emit("timerEnded"));
+</script>
+
+<template>
+  <time
+    class="onyx-toast-progress-bar"
+    role="progressbar"
+    :aria-valuemin="0"
+    :aria-valuemax="props.duration"
+    :aria-valuenow="percentage"
+    :aria-label="t('toast.progressBarLabel')"
+    :style="{ width: `${percentage}%` }"
+  ></time>
+</template>
```

Comment:
Great implementation! Unfortunately, this seems to be an impractical pattern for accessibility devices.
Instead, the notification text should be announced via a live-region. The screen-reader device should not be aware of actual toast and it's duration.